src/App.tsx

```diff
@@ -15,71 +16,82 @@ export const goods = [
-      />
-    </h1>
+  handleClearButton = () => {
+    this.setState({ selectedGood: '' });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  handleAddRemoveButton = (good: string) => {
+    this.setState((state) => (
+      { selectedGood: (state.selectedGood === good) ? '' : good }
+    ));
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        <h1 className="title is-flex is-align-items-center">
+          {selectedGood
+            ? `${selectedGood} is selected`
+            : 'No goods selected'}
 
-        <tr data-cy="Good">
-          <td>
+          {selectedGood && (
+            /* eslint-disable-next-line jsx-a11y/control-has-associated-label */
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              className="delete ml-3"
+              onClick={this.handleClearButton}
+            />
+          )}
+        </h1>
+        <table className="table">
+          <tbody>
+            {goods.map(good => (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={classNames(
+                  { 'has-background-success-light': selectedGood === good },
+                )}
+              >
+                <td>
+                  <button
+                    data-cy={selectedGood === good
+                      ? 'RemoveButton'
+                      : 'AddButton'}
+                    type="button"
+                    className={classNames(
+                      'button',
+                      { 'is-info': selectedGood === good },
+                    )}
+                    onClick={() => {
+                      this.handleAddRemoveButton(good);
+                    }}
+                  >
+                    {selectedGood === good
+                      ? '-'
+                      : '+'}
+                  </button>
```

Comment:
also, it is hard to read(( It is better to make two separate button and show one of them depending in condition